### ISSUE TITLE
Proxyquire mutates stub if overridable module exports an array

### ISSUE THREAD

**Issue by everm1nd** [thlorenz/proxyquire#191]
Expected:
`goodies` dependency is being overloaded my proxyquire and replaced with `stubs` variable.

Actual:
`stubs` is mutated and contains one additional element from `goodies` module. `someModule` gets this mixed array as an overload.

```js
// --- someModule.test.js

const stubs = [
  'oneStub',
  'anotherStub'
];

const someModule = proxyquire('./someModule', {
  './goodies': stubs
});

it('keeps stubs untouched', () => {
  // fails here: "AssertionError: expected [ 'oneStub', 'anotherStub', 'C' ]
  // to deeply equal [ 'oneStub', 'anotherStub' ]"
  expect(stubs).to.eql(['oneStub', 'anotherStub']);
});
```

```js
// --- someModule.js

const goodies = require('./goodies');

module.exports = goodies;
```

```js
// --- goodies.js

module.exports = [
  'A',
  'B',
  'C'
];
```

**Comment by bendrucker** [comment]
We do not clone stubs like you're expecting. That test will never pass. The asserted value should include A, B, C.

**Comment by bendrucker** [comment]
Hey, I tested this on the upstream package that tries to merge the keys. My original expectation was wrong—see those two linked commits for the behavior. I was unable to reproduce your specific problem where you see only once element appended. 

There's no clear right answer here for arrays. Stub objects are merged with the actual module by default as a convenience (e.g. patch `fs`) but I'd rather not get into merging two populated arrays. 

I've added tests both upstream and in proxyquire clarifying what the actual behavior is regarding arrays specifically. In particular see the notes in d20312e. Happy to take additional contributions here.
